README.md

```diff
@@ -68,6 +68,168 @@ You can serve your Connect RPCs with vanilla Node.js, or use our [server plugins
+- We have removed the `credentials` option from transports as well as the `init`
+  option in interceptors. These two options were used to customize `fetch` routines.
+  Users should now rely on the `fetch` option in transports as a way to perform
+  these customizations. For example:
+
+  ```ts
+  createConnectTransport({
+    baseUrl: "/",
+    fetch: (input, init) => fetch(input, { ...init, credentials: "include" }),
+  });
+  ```
+
+  In addition, as a replacement to determine whether an incoming request is a
+  Connect GET request in server-side interceptors, the property `requestMethod: string`
+  has been added to intercepted requests. This property is symmetrical to `HandlerContext.requestMethod`.
+
+- The access pattern of `MethodDescriptor` has slightly changed. In v2, the
+  Protobuf-ES plugin generates service descriptors and changes the
+  access pattern for method descriptors. Instead of the plural form `methods`,
+  it now uses the singular `method`.
+
+  For example: `ElizaService.methods.Say` now becomes `ElizaService.method.Say`.
+
+- Errors details are now a pair of desc and init values. In Connect v1, error
+  details were specified as message instances. In v2, error
+  details are now an object that specifies both a schema and initialization
+  object which are both passed to the `create` function of Protobuf-ES. For example:
+
+  ```diff
+  - import { LocalizedMessage } from "./gen/google/rpc/error_details_pb";
+  - const details = [
+  -   new LocalizedMessage({
+  -       locale: "fr-CH",
+  -       message: "Je n'ai plus de mots.",
+  -   }),
+  - ];
+  + import { LocalizedMessageSchema } from "./gen/google/rpc/error_details_pb";
+  + const details = [
+  +   {
+  +     desc: LocalizedMessageSchema,
+  +     value: {
+  +       locale: "fr-CH",
+  +       message: "Je n'ai plus de mots.",
+  +     }
+  +   },
+  + ];
+
+  const metadata = new Headers({
+    "words-left": "none"
+  });
+  throw new ConnectError(
+    "I have no words anymore.",
+    Code.ResourceExhausted,
+    metadata,
+    details
+  );
+  ```
+
+- `MethodDescriptor` is now self-sufficient. In v1, method descriptors always
+  had to be used alongside the service descriptors. In v2, they can now be used
+  standalone. This means that all callsites that previously passed a service
+  and method descriptor can now just pass the method descriptor.
+
+  A notable example is when using a Connect `Router` on the server-side:
+
+  ```diff
+  const routes = ({rpc}: Router) => {
+  -   rpc(ElizaService, ElizaService.say, impl);
+  +   rpc(ElizaService.say, impl);
+  }
+  ```
+
+- Interceptors for streaming RPCs now use appropriate stream types. In
+  v1, the server used more exact types in interceptors, for example `UnaryRequest`
+  for server-streaming rpcs while the client always used streaming variants.
+  This was unintended behavior and has been fixed in v2. Now all streaming RPCs
+  use the `StreamRequest` and `StreamResponse` types on the server as well.
```

Comment:
This probably could use a code example, but wasn't sure of the best way to illustrate it.